### PR TITLE
Add solution filter for shared tests

### PR DESCRIPTION
**Pull request by mrlacey**
to be used in the PR pipeline
For #4438

# PR checklist

**Quick summary of changes**

**Which issue does this PR relate to?**

**Applies to the following platforms:**

- [ ] WinUI
- [ ] WPF
- [ ] UWP

**Anything that requires particular review or attention?**

**Do all automated tests pass?**

**Have automated tests been added for new features?**

**If you've changed the UI:**
  - Be sure you are including screenshots to show the changes.
  - Be sure you have reviewed the [accessibility checklist](accessibility.md).

**If you've included a new template:**
  - Be sure you reviewed the [Template Verification Checklist](https://github.com/microsoft/TemplateStudio/wiki/Checklist:-Template-Verification).
  - Be sure it's included in the list on this [UWP](https://github.com/microsoft/TemplateStudio/blob/main/docs/UWP/getting-started-endusers.md) or [WPF](https://github.com/microsoft/TemplateStudio/blob/main/docs/WPF/getting-started-endusers.md) getting started doc.

**Have you raised issues for any needed follow-on work?**

**Have docs been updated?**

**If breaking changes or different ways of doing things have been introduced, have they been communicated widely?**

### CODE REVIEW

**Comment by mrlacey** [comment]
That's an even bigger improvement than I was expecting!

Although these jobs may be running in parallel and so the impact of this change may be reduced.

Need to allow for some variations in build times on the server (because of load & _"reasons"_) but even still this is better.
Less time waiting for checks to complete. 😃 


**Before** (random sample):

13m 13s & 10m 18s

**After** (build for this PR):

4m 15s & 3m 24s

**Comment by mrlacey** [comment]
This [failed on one attempt](https://winappstudio.visualstudio.com/WTS/_build/results?buildId=38340&view=logs&j=37d10d5d-e62d-5f4f-b769-ae2539dbd684&t=532deab7-1da9-5ec5-b782-b692fc07a8c3) due to an intermittent error with the nuget restore task.
@mikebattista how do you feel about adding automatic retries for things like this?

**Comment by mikebattista** [comment]
Conditional execution of build/test based on what changed we can look at after we ship the initial VS 2022 versions.

What we need for that, though, is a spec for what changes in the repo require what platforms and tests to be run.

**Comment by mikebattista** [comment]
You know, another quick win here would be to not run the UWP tests for CI given that (1) we don't expect many changes there going forward and (2) it takes the majority of the time (1 hour+ compared to 5-20 minutes for everything else).

If we assume the changes to UWP will be infrequent and that the nightly test passes would catch anything egregious, we can disable that from CI.

**Comment by mikebattista** [comment]
> @mikebattista Why change the order of the Shared matrix after you reverted the Test order change?

It's purely cosmetic.

File: _build/pipelines/ci.yml [hunk 1]
```diff
@@ -102,8 +102,8 @@ jobs:
         Moniker: WPF
         BuildConfiguration: DebugWPF
         BuildPlatform: Any CPU
-      # Reenable as part of #4441
-      #UWP:
+      # #4441
```
Comment (mrlacey) [inline _build/pipelines/ci.yml:105]: Why remove additional context for this issue number?
If all I see is an issue number and I don't remember why it's there (I won't,) if I'm looking at this file and wonder about this commented out section I then have to go and look at the referenced issue to see why that's the case.
Does the referenced issue explain why it's commented out? Does it explain when or why to reenable it? Is it important? Is it just there for information purposes only? Does it directly relate to the commented-out lines below?

I'm all for standardizing formatting and having after the character that indicates a comment, but removing contextual information that is intended to aid in future code maintenance (for the sake of "cosmetic" improvements) doesn't feel like a useful change.

**Not just me but anyone who looks at this file.** - Others are even more likely to need to go and look this up.

Comment (mikebattista) [reply _build/pipelines/ci.yml:105]: "Reenable as part of #4441" also isn't enough context to answer the questions you raised. If you think looking at commented lines here is confusing, I'd rather just remove the whole UWP section, but I prefer the way it is now with comments and an issue number for reference.

**Review by mikebattista — APPROVED** [review]

**Pull request merged**